extension/encoding/awscloudwatchmetricstreamsencodingextension/extension.go

```diff
@@ -14,21 +15,30 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding"
 )
 
+var errEmptyRecord = errors.New("0 metrics were extracted from the record")
+
 var _ encoding.MetricsUnmarshalerExtension = (*encodingExtension)(nil)
 
 type encodingExtension struct {
-	pmetric.Unmarshaler
+	unmarshaler pmetric.Unmarshaler
+	format      string
 }
 
 func newExtension(cfg *Config, settings extension.Settings) (*encodingExtension, error) {
 	switch cfg.Format {
 	case formatJSON:
-		return &encodingExtension{&formatJSONUnmarshaler{
-			buildInfo: settings.BuildInfo,
-			logger:    settings.Logger,
-		}}, nil
+		return &encodingExtension{
+			unmarshaler: &formatJSONUnmarshaler{
+				buildInfo: settings.BuildInfo,
+				logger:    settings.Logger,
+			},
+			format: formatJSON,
```

Comment:
Not a strong opinion, but would be ideal to pass in config instead of the format.

Reply:
I am not convinced that it is a good idea if it is only for one value you want to pass. 

Storing the config (or reference) does have some memory impact considerations, and it does encourage some amount of laziness from what I have seen historically.

Reply:
Agreed - I'd rather pass down the minimal amount of config/context necessary as we're currently doing.